tests/core/test_model.py

```diff
@@ -290,6 +291,32 @@ def test_model_download(tmpdir, cls, filename):
         assert isinstance(task, cls)
 
 
+class DummyTask(Task):
+    def __init__(self):
+        super().__init__()
+
+        self.model = nn.Sequential(
+            nn.Linear(10, 20),
+            nn.Linear(20, 30),
+            nn.Linear(30, 40),
+        )
+
+    def predict_step(self, batch, batch_idx: int, dataloader_idx: int = 0):
+        return self.model(batch)
+
+
+def test_as_embedder():
+    embedder = DummyTask().as_embedder("model.1")
+
+    assert isinstance(embedder, Embedder)
+    assert embedder.predict_step(torch.rand(10, 10), 0, 0).size(1) == 30
```

Comment:
Minor suggestion: probably better if we don't hard-code, and use `out_features` from the respective layer:

```suggestion
def test_as_embedder():
    layer_number = "1"
    embedder = DummyTask().as_embedder("model." + layer_number)

    assert isinstance(embedder, Embedder)
    assert (
        embedder.predict_step(torch.rand(10, 10), 0, 0).size(1)
        == embedder.model.model._modules[layer_number].out_features
    )
```